app/src/main/java/com/iic/shopingo/ui/request_flow/activities/SaveRequestActivity.java

```diff
@@ -21,16 +27,23 @@ protected void onCreate(Bundle savedInstanceState) {
 
     Contact shopper = getIntent().getParcelableExtra(EXTRAS_SHOPPER_KEY);
     ShoppingList shoppingList = getIntent().getParcelableExtra(EXTRAS_SHOPPING_LIST_KEY);
-    OutgoingRequest request = saveRequest(shopper, shoppingList);
-    Intent intent = new Intent(this, RequestStateActivity.class);
-    intent.putExtra(RequestStateActivity.EXTRAS_REQUEST_KEY, request);
-    startActivity(intent);
-    finish();
-  }
 
-  // TODO: Move this to somewhere more fitting
-  private OutgoingRequest saveRequest(Contact shopper, ShoppingList shoppingList) {
-    // TODO: Send `shopper` and `shoppingList` to server and get an `OutgoingRequest` back
-    return new OutgoingRequest(shopper, shoppingList, BaseRequest.RequestStatus.PENDING);
+    MakeRequest req = new MakeRequest(CurrentUser.getToken(), shoppingList.getItems(), shoppingList.getOffer(), shopper.getId());
+    req.executeAsync().continueWith(new Continuation<OutgoingRequestApiResult, Object>() {
```

Comment:
I think we can do away with this activity, and have this code in the `CreateShoppingListActivity` when the user clicks on `Send request`.
Even in a scenario where the user gets his request declined and selects another shopper we want to show him the `CreateShoppingListActivity` with all the values pre-filled, so he'll tap on the `Send request` button again.